var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -0,0 +1,197 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+#
+# Please also see the LICENSE file for our notice and the LGPL.
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+import string
+
+
+class Lammps(Package):
+    """LAMMPS stands for Large-scale Atomic/Molecular Massively
+    Parallel Simulator."""
+    homepage = "http://www.example.com"
+    url      = "https://github.com/lammps/lammps/archive/stable_17Nov2016.tar.gz"
+
+    version('17Nov2016', '8aecc58a39f9775203517c62a592d13b',
+            url="https://github.com/lammps/lammps/archive/stable_17Nov2016.tar.gz")
+
+    supported_packages = ['voronoi', 'rigid', 'user-nc-dump',
+                          'user-atc', 'meam', 'manybody']
```

Comment:
In  fact this are the packages that our local users reported, all the packages in lammps seem to have a bit they own particularities, so appart if there is a strong need for some particular ones I did not really plan to go into all of them.

The part that I mentioned about varant that I do not know how to handle is that 'manybody' can be used alone but 'user-atc' needs many body to be activated. Going for a multi-variant with #2386 might help since it allow to check conditions on the variant content